src/Routers/ClassesRouter.js

```diff
@@ -187,11 +187,11 @@ export class ClassesRouter extends PromiseRouter {
     if (body.count) {
       options.count = true;
     }
-    if (typeof body.keys == 'string') {
-      options.keys = body.keys;
+    if (body.keys != null) {
+      options.keys = String(body.keys);
     }
-    if (typeof body.excludeKeys == 'string') {
-      options.excludeKeys = body.excludeKeys;
+    if (body.excludeKeys) {
```

Comment:
Should we change this also to `== null`?

Reply:
Did you mean `!= null`? If so, aren't these the same thing as `if (body.excludeKeys)`? I know javascript is loosely typed so this might mean something I don't understand.

Reply:
I mean since we are changing to `keys != null` shouldn't we also change `excludedKeys != null` so that we don't run into the same issue for `exclude([])` as we do here for `select([])`?

So, `!= null` in JS is true for any value except `undefined` and `null`.

Reply:
I don't know if there is a `exclude([])` as I didn't see it in the test cases I looked at and never tried it.

Reply:
I think if it was there originally, it would have thrown a test failure when I first submitted the PR as the `select([])` did throw a failure

Reply:
There's no test for exclude([]) in ParseQuery.spec.js, so I don't know what will happen if someone tries it

Reply:
I mentioned that because if we have a different condition for select vs. include, we may create an inconsistency in the developer experience.

Maybe we should add the missing case for `exclude([])`?

Reply:
Are you sure `exclude([])` will have the behavior you expect? If so what's the behavior, something like "exclude all"?

Reply:
exclude([]) should do nothing. I don’t know why I didn’t add that test case.

Reply:
@dplewis like nothing at all? If so, this seems to be the results I see from the test, it returns the whole Parse Object

Reply:
It gets ignored, all keys will be returned

Reply:
So then that "doesn't" need a test then right? Or should I add it?

Reply:
I just realized there is a performance improvement that could be made here with exclude. I’ll add it to my schema improvement PR

Reply:
I have a testcase for this, should I add it or leave it your PR?

```javascript
const response3 = await request({
      url: Parse.serverURL + '/classes/TestObject',
      qs: {
        excludeKeys: [],
        where: JSON.stringify({ objectId: obj.id }),
      },
      headers: masterKeyHeaders,
    });
    ok(response3.data.results[0].objectId, 'expected objectId to be set');
    ok(response3.data.results[0].createdAt, 'expected object createdAt to be set');
    ok(response3.data.results[0].updatedAt, 'expected object updatedAt to be set');
    expect(response3.data.results[0].foo).toBe('baz');
    expect(response3.data.results[0].hello).toBe('world');
```

Reply:
> So then that "doesn't" need a test then right? Or should I add it?

It needs a test because `exclude([])` is an allowed and syntactically valid value, just like `select([])`, see the list of extrema above.

Reply:
You should add it, it wouldn’t hurt

Reply:
I think the test case should be done with the JS SDK commands, because the REST request basically shortcuts the JS SDK, but we want to make sure it works with JS SDK.

Reply:
I added the test... @mtrezza if I understood you correctly earlier, I don't think `exclude([])` does what you expected. The line @dplewis pointed out, https://github.com/parse-community/parse-server/blob/master/src/RestQuery.js#L712, basically treats `exclude([])` as if was never there which is different from `select([])`

Reply:
> I think the test case should be done with the JS SDK commands, because the REST request basically shortcuts the JS SDK, but we want to make sure it also works with JS SDK.

Done, and I added a REST test for `select` as the only time it was tested with REST were in the `exclude` tests.

Reply:
> I don't think exclude([]) does what you expected. 

I does what I expected, it should do nothing, but again, that is different from `select([])` which should do something.